### ISSUE TITLE
Option to ignore var in for-loops in no-inner-declarations?

### ISSUE THREAD

**Issue by goloroden** [eslint/eslint#976]
Basically, I love the rule `no-inner-declarations` - except for one thing: While I strongly agree that it's a bad idea to declare variables _somewhere_, I usually declare `i` as index of a loop inside the loop definition itself, such as:

``` javascript
for (var i = 0; i < 10; i++) {
  // …
}
```

However, I can't have both at the moment: Either I'm allowing declaration of variables _everywhere_ - or I deny it. What I would like to have is the option to tell: Deny it everywhere, _except_ for loop variables.

What do you think about this?
